tests/python_package_test/test_dask.py

```diff
@@ -377,6 +377,27 @@ def test_find_random_open_port(client):
     client.close(timeout=CLIENT_CLOSE_TIMEOUT)
 
 
+def test_worker_map_has_duplicates():
+    map_with_duplicates = worker_map = {
+        'tcp://127.0.0.1:8786': 123,
+        'tcp://127.0.0.1:8788': 123,
+        'tcp://10.1.1.2:15001': 123
+    }
+    assert lgb.dask._worker_map_has_duplicates(map_with_duplicates)
+
+    map_without_duplicates = {
+        'tcp://127.0.0.1:8786': 12405,
+        'tcp://10.1.1.2:15001': 12405
+    }
+    assert lgb.dask._worker_map_has_duplicates(map_without_duplicates) is False
+
+    localcluster_map_without_duplicates = {
+        'tcp://127.0.0.1:708': 12405,
+        'tcp://127.0.0.1:312': 12405,
+    }
+    assert lgb.dask._worker_map_has_duplicates(map_without_duplicates) is False
```

Comment:
```
# Correct:
if greeting:
```
```
# Wrong:
if greeting is True:
```

   https://www.python.org/dev/peps/pep-0008/#programming-recommendations

```suggestion
    assert not lgb.dask._worker_map_has_duplicates(map_without_duplicates)

    localcluster_map_without_duplicates = {
        'tcp://127.0.0.1:708': 12405,
        'tcp://127.0.0.1:312': 12405,
    }
    assert not lgb.dask._worker_map_has_duplicates(map_without_duplicates)
```

Reply:
I disagree that what I've written is "wrong" and what you're proposing is "correct".

I am trying to strictly test that `_worker_map_has_duplicates()` returns a specific boolean value, not a "true-ish" or "false-ish" value.

All of these conditions would pass the pattern you're proposing:

```python
assert not []
assert not ""
assert not set()
assert not {}
```

Reply:
It's not me, it's PEP8. But I understand your intention.
On the other side, `_worker_map_has_duplicates()` is designed to return a boolean value which is reflected in its type annotation: `-> bool`.

Reply:
this code is passing our linters, which I understood to be our agreement about style for Python code in this project.

After hearing my response, do you still want me to change this to remove the `is`?

Reply:
This case is a little bit beyond just linting rules.
But let's leave it as `is False`, OK.

Reply:
alright, thanks